---
title: "HLAtools.data"
author: 
- Ryan M. Nickens -- rynickens@gmail.com
- Livia Tran -- livia.tran@ucsf.edu
- Leamon Crooms -- lcroomsiv@arizona.edu
- Derek Pappas, PhD -- djpappas75@gmail.com
- Vinh Luu -- vinhluu864@berkeley.edu
- Josh Bredeweg
- Steven J. Mack, PhD -- steven.mack@ucsf.edu
date: "2024-02-08"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Using HLAtools.data}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
options(rmarkdown.html_vignette.check_title = FALSE)
```

-   Package Version: 0.5.2.9000

# Overview

The *HLAtools.data* package contains the *HLAalignments* and *HLAatlas* objects. These are refereence objects on their own, but are also required for functions in the *HLAtools* package. These data objects are created by the *HLAtools::alignmentFull()* and *HLAtools::atlasFull()* functions.

## Data and Formats
### Alignments
The *HLAalignments* data object is a list object containing sub-lists of peptide (*prot*), coding nucleotide (*nuc*), and genomic nucleotide (*gen*) lists, along with a *version* character string that identifies the IPD-IMGT/HLA Database release under which the *HLAalignments* object was built. 

Each of these sub-lists contains data frames of the respective IPD-IMGT/HLA Database alignment for each locus in IPD-IMGT/HLA Database release 3.55.0. These protein, nucleotide and genomic DNA sequence alignments are described in detail below.

#### Protein Aligments
The column header for the *prot* alignments identifies individual peptide positions, starting from the first peptide of the leader sequence (a negative position) or the start codon, when there is no leader sequence. For example, as shown in Table 1, the HLA-A *prot* alignment begins at codon position -24.

Table 1. Headers and Amino Acid Resideue Position Numbering in the HLA-A Nucleotide Alignment

|locus|allele|trimmed_allele|allele_name|-24|-23|-22|-21|-20|-19|
|:-----:|:----------:|:------------:|:-------------:|:-:|:---:|:---:|:-:|:---:|:---:|
|A|01:01:01:01|A*01:01|A*01:01:01:01|M|A|V|M|A|P|

#### Nucleotide (cDNA) Sequence Alignments
The column heads for the *nuc* alignments identify the individual nucleotide positions within each codon, starting with the first codon. As shown in Table 2, the HLA-A *nuc* alignment begins at the first nucleotide in codon position -24, which is identified as *-24*; the second nucleotide in codon -24 is identified as *-24.1* and the third nucleotide in codon -24 is identified as *-24.2*. 

Table 2. Headers and Codon Position Numbering in the HLA-A Nucleotide Alignment

|locus  |allele      |trimmed_allele|allele_name    |-24|-24.1|-24.2|-23|-23.1|-23.2|
|:-----:|:----------:|:------------:|:-------------:|:-:|:---:|:---:|:-:|:---:|:---:|
|A      |01:01:01:01 |  A\*01:01    | A\*01:01:01:01| A | T   | G   | G | C   | C   |

#### Genomic (gDNA) Sequence Alignments
The column heads for the *gen* alignments identify the individual nucleotide positions in the 'full gene' sequence, which usually starts in the 5' untranslated region (UTR) sequence for expressed genes 

Table 3. Headers and Nucleotide Position Numbering in the HLA-A Genomic Alignment

|locus|allele     |trimmed_allele|allele_name   |-300|-299|-298|-297|-296|-295|
|:-----:|:----------:|:------------:|:-------------:|:-:|:---:|:---:|:-:|:---:|:---:|
|A    |01:01:01:01|A\*01:01      |A\*01:01:01:01|C   |A   |G   |G   |A   |G   |


### Atlases
The *HLAatlas* data object is a list object containing sub lists of data frames (atlases) for each locus with a *prot*, *nuc* or *gen* alignment, along with a *version* character string that identifies the IPD-IMGT/HLA Database release under which the *HLAatlas* object was built. Each atlas identifies the position of the gene feature (exon, intron or untranslated region [UTR]) boundaries in the respective alignment type for each gene. In nucleotide alignment atlases, the value describing the location of the boundary between two features refers to the nucleotide position on the **3' side** of the boundary. In most peptide alignment atlases, the value describing the location between two features refers to the **codon** in which the boundary is found. The *TAP1* and *TAP2* genes contain a mixture of boundaries that occur within and between codons; these are not distinguished in the atlas. 

For example, the *HLAtools.data\:\:HLAatlas\$prot\$A* atlas identifies the Exon 1:Exon 2 boundary as occurring "within" HLA-A amino acid (AA) 1 (Table 4).

Table 4. HLA-A Protein Atlas

| |E.1-2|E.2-3|E.3-4|E.4-5|E.5-6|E.6-7|E.7-8|
|:------:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
|AA      |  1  |  91 | 183 | 275 | 314 | 325 | 341 |

This is reflected in the *HLAtools.data\:\:HLAatlas\$nuc\$A* atlas (Table 5), which identifies the Exon 1: Exon 2 boundary as occurring before cDNA position 65 and within codon 1. The *HLAtools.data\:\:HLAaligments\$nuc\$A* alignment indicates that the boundary is between codon 1 nucleotides 1 and 2.  

Table 5. HLA-A Nucleotide Atlas

| |E.1-2|E.2-3|E.3-4|E.4-5|E.5-6|E.6-7|E.7-8|
|:------:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
|cDNA    |  74 |344  |  620|  896|1013 |1046 | 1094|
|codon   | 1   |91   |183  |275  |314  |325  |341|

Finally, the *HLAtools.data\:\:HLAatlas\$gen\$A* atlas identifies the location of all feature boundaries in the genomic alignment. As shown in Table 6, the boundary between the 5' UTR and exon 1 is between gDNA positions -1 and +1. 

Table 6. HLA-A Genomic Altas

| |U.5-E.1|E.1-I.1|I.1-E.2|E.2-I.2|I.2-E.3|E.3-I.3|I.3-E.4|E.4-I.4|I.4-E.5|E.5-I.5|I.5-E.6|E.6-I.6|I.6-E.7|E.7-I.7|I.7-E.8|E.8-U.3|
|:------:|:---:|:---:|:---:|:---:|:---:|:---:|:------:|:---:|:---:|:---:|:---:|:---:|:---:|:------:|:---:|:---:|
|gDNA    |1      |74     |204    |474    |715    |991    |1570   |1846   |1948   |2065   |2507   |2540   |2682   |2730   |2899   |2904   |

##

End of vignette.
